examples/summarization/bart/run_bart_sum.py

```diff
@@ -0,0 +1,247 @@
+import argparse
+import glob
+import logging
+import os
+import random
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+
+from transformers import AdamW, BartTokenizer, get_linear_schedule_with_warmup
+from transformers.modeling_bart import BartForConditionalGeneration
+from utils import CnnDailyMailDataset, add_generic_args
+
+
+logger = logging.getLogger(__name__)
+
+
+class BartSystem(pl.LightningModule):
+    def __init__(self, hparams):
+        super(BartSystem, self).__init__()
+        self.hparams = hparams
+        self.bart = BartForConditionalGeneration.from_pretrained("bart-large", output_past=True)
+
```

Comment:
This will eventually be handled by the transformer_base.py in #3290 , but for now can you add a config object in this `__init__`? Something like this:

```python
        config = BartConfig.from_pretrained(
            self.hparams.config_name if self.hparams.config_name else self.hparams.model_name_or_path,
            num_labels=num_labels,
            cache_dir=self.hparams.cache_dir if self.hparams.cache_dir else None,
        )
        tokenizer = BartTokenizer.from_pretrained(
            self.hparams.tokenizer_name if self.hparams.tokenizer_name else self.hparams.model_name_or_path,
            do_lower_case=self.hparams.do_lower_case,
            cache_dir=self.hparams.cache_dir if self.hparams.cache_dir else None,
        )
        model = BartForConditionalGeneration.from_pretrained(
            self.hparams.model_name_or_path,
            from_tf=bool(".ckpt" in self.hparams.model_name_or_path),
            config=config,
            cache_dir=self.hparams.cache_dir if self.hparams.cache_dir else None,
        )
        self.config, self.tokenizer, self.model = config, tokenizer, model
```

Reply:
I've made those changes, it looks like  #3290 has been merged now, how will this change?

Reply:
So following https://github.com/huggingface/transformers/pull/3290 we would like you to use a shared base class for BartSystem. This should eliminate your `constructor`, `optimizer_step`, `configure_optimizers`, and several other functions.